pkg/tcpip/network/ipv4/ipv4.go

```diff
@@ -306,6 +317,17 @@ func (e *endpoint) HandlePacket(r *stack.Route, vv buffer.VectorisedView) {
 	vv.TrimFront(hlen)
 	vv.CapLength(tlen - hlen)
 
+	// If options are included in incoming packet, store it into byte array.
+	if hlen > header.IPv4MinimumSize {
+		options = h.Options(hlen - header.IPv4MinimumSize)
+		if len(options) != int(options[1]) || len(options) <= int(options[3]) {
+			e.ParamProblem(r, headerView, vv)
+			return
+		}
+	} else {
+		options = nil
+	}
```

Comment:
nil assignment is not required.

Reply:
here options is an global variable, once options get set in any of the incoming packets it will be available for all the packets, then it will affect in writePacket of ipv4 so I believe it is required.